src/core/Akka.Tests/IO/TcpIntegrationSpec.cs

```diff
@@ -416,91 +416,91 @@ public async Task Should_fail_writing_when_buffer_is_filled()
 
                 actors.ServerHandler.Send(actors.ServerConnection, Tcp.Write.Create(ByteString.FromBytes(new byte[100000]), Ack.Instance));
-                actors.ServerHandler.ExpectMsg(Ack.Instance);
+                await actors.ServerHandler.ExpectMsgAsync(Ack.Instance);
 
-                x.ExpectReceivedData(actors.ClientHandler, 100000);
+                await x.ExpectReceivedDataAsync(actors.ClientHandler, 100000);
             });
         }
 
         [Fact]
-        public void The_TCP_transport_implementation_dont_report_Connected_when_endpoint_isnt_responding()
+        public async Task The_TCP_transport_implementation_dont_report_Connected_when_endpoint_isnt_responding()
         {
             var connectCommander = CreateTestProbe();
             // a "random" endpoint hopefully unavailable since it's in the test-net IP range
             var endpoint = new IPEndPoint(IPAddress.Parse("192.0.2.1"), 23825);
             connectCommander.Send(Sys.Tcp(), new Tcp.Connect(endpoint));
             // expecting CommandFailed or no reply (within timeout)
-            var replies = connectCommander.ReceiveWhile(TimeSpan.FromSeconds(1), x => x as Tcp.Connected);
+            var replies = await connectCommander.ReceiveWhileAsync(TimeSpan.FromSeconds(1), x => x as Tcp.Connected).ToListAsync();
             replies.Count.ShouldBe(0);
         }
 
         [Fact]
-        public void Should_report_Error_only_once_when_connecting_to_unreachable_DnsEndpoint()
+        public async Task Should_report_Error_only_once_when_connecting_to_unreachable_DnsEndpoint()
         {
             var probe = CreateTestProbe();
             var endpoint = new DnsEndPoint("fake", 1000);
             Sys.Tcp().Tell(new Tcp.Connect(endpoint), probe.Ref);
             
             // expecting CommandFailed or no reply (within timeout)
-            var replies = probe.ReceiveWhile(TimeSpan.FromSeconds(5), x => x as Tcp.CommandFailed);
+            var replies = await probe.ReceiveWhileAsync(TimeSpan.FromSeconds(5), x => x as Tcp.CommandFailed).ToListAsync();
             replies.Count.ShouldBe(1);
         }
 
         [Fact]
-        public void The_TCP_transport_implementation_handle_tcp_connection_actor_death_properly()
+        public async Task The_TCP_transport_implementation_handle_tcp_connection_actor_death_properly()
         {
-            new TestSetup(this, shouldBindServer:false).Run(x =>
+            await new TestSetup(this, shouldBindServer:false).RunAsync((Action<TestSetup>)(async x =>
             {
                 var serverSocket = new Socket(SocketType.Stream, ProtocolType.Tcp);
                 serverSocket.Bind(new IPEndPoint(IPAddress.Loopback, 0));
                 serverSocket.Listen(100);
                 var endpoint = (IPEndPoint) serverSocket.LocalEndPoint;
 
-                var connectCommander = CreateTestProbe();
+                var connectCommander = base.CreateTestProbe();
```

Comment:
Why do you need to call base?

Reply:
Ah, I don't remember touching that though!